lib/contracts/contribution.js

```diff
@@ -48,12 +63,15 @@ class Contribution extends Base {
   }
 
   addContribution(contributionAttr, callOptions = {}) {
-    let json = ContributionSerializer.serialize(contributionAttr);
-    // TODO: validate against schema
+    let jsonStr = ContributionSerializer.serialize(contributionAttr);
+
+    // Validate JSON document against schema
+    let result = validator.validate(JSON.parse(jsonStr), schemas['contribution']);
```

Comment:
I somewhat dislike that we do a `JSON.stringify` in the ContributionSerializer and then do a `JSON.parse` again to validate it. 
maybe we can have two function in the ContributionSerializer? 
of even do the validation in the ContributionSerializer? 

something like: 
```js
const c = Contribtion.new(contribtionAttr);
c.isValid() // calls the validator.validate() function
c.error // returns validator.error
c.serialize() // same as now
```

but we can refactor that later I guess.

Reply:
Yes, makes sense. We can refactor it when adding validation for contributor details.

Reply:
Maybe something like:

```js
// This tv4 api feels strange...
const [valid, validator] = ContributionSerializer.validate(contribtionAttr);
if (!valid) { return Promise.reject(validator.error); }
```
`.validate` should use `_serialize()` to do the work and `serialize()` can use the private method as well but returns the JSON string...

I think the overhead of `_serialize()` is not much so we can do it twice.

Reply:
Already came up with an even cleaner solution. Promisify ALL THE THINGS. ;)

Reply:
https://github.com/67P/kredits-contracts/blob/d953141f52d515dbcba4216c5eab9111c58467a2/lib/contracts/contribution.js#L51-L54